drivers/kw2xrf/kw2xrf.c

```diff
@@ -0,0 +1,1041 @@
+        dev->buf[index++] = (ng_netif_hdr_get_dst_addr(hdr))[0];
+        /* set source pan_id */
+        dev->buf[index++] = (uint8_t)((dev->radio_pan) >> 8);
+        dev->buf[index++] = (uint8_t)((dev->radio_pan) & 0xff);
+        /* set source address */
+        dev->buf[index++] = (uint8_t)(dev->addr_short[0]);
+        dev->buf[index++] = (uint8_t)(dev->addr_short[1]);
+    }
+
+    if (hdr->dst_l2addr_len == 8) {
+        /* set destination pan_id, wireshark expects it there */
+        dev->buf[index++] = (uint8_t)((dev->radio_pan) >> 8);
+        dev->buf[index++] = (uint8_t)((dev->radio_pan) & 0xff);
+        /* default to use long address mode for src and dst */
+        dev->buf[2] |= 0xcc;
+        /* set destination address located directly after ng_ifhrd_t in memory */
+        memcpy(&(dev->buf)[index], ng_netif_hdr_get_dst_addr(hdr), 8);
+        index += 8;
+        /* set source pan_id, wireshark expects it there */
+        dev->buf[index++] = (uint8_t)((dev->radio_pan) >> 8);
+        dev->buf[index++] = (uint8_t)((dev->radio_pan) & 0xff);
+
+        /* set source address */
+        memcpy(&(dev->buf[index]), dev->addr_long, 8);
+        index+=8;
+    }
+    return index;
+}
+
+
+int _send(ng_netdev_t *netdev, ng_pktsnip_t *pkt)
+{
+    uint8_t index=0;
+    kw2xrf_t *dev = (kw2xrf_t*) netdev;
+    ng_pktsnip_t *payload = pkt->next;
+
+    if (pkt == NULL) {
+        return -ENOMSG;
+    }
+    if (netdev == NULL) {
+        ng_pktbuf_release(pkt);
+        return -ENODEV;
+    }
+
+    if (pkt->type == NG_NETTYPE_NETIF) {
+        /* Build header and fills this already into the tx-buf */
+        index = _assemble_tx_buf(dev, pkt);
+        DEBUG("Assembled header for NG_NETTYPE_UNDEF to tx-buf, index: %i\n", index);
+    }
+    else if (pkt->type == NG_NETTYPE_UNDEF) {
+        /* IEEE packet is already included in the header,
+         * no need to build the header manually */
+        DEBUG("Incoming packet of type NG_NETTYPE_802154: %i\n", index);
+        DEBUG("size of pktsnip: %i\n", pkt->size);
+        for (int i=0; i < pkt->size; i++) {
+            uint8_t *tmp = pkt->data;
+            dev->buf[index+i+1] = tmp[i];
+        }
+        /* count bytes */
+        index += pkt->size;
+    }
+    else {
+        DEBUG("Driver does not support this type of packet\n");
+        return -ENOTSUP;
+    }
+
+    while (payload) {
+        /* check we don't exceed FIFO size */
+        if (index+2+payload->size > KW2XRF_MAX_PKT_LENGTH) {
+            ng_pktbuf_release(pkt);
+            DEBUG("Packet exceeded FIFO size.\n");
+            return -ENOBUFS;
+        }
+        for (int i=0; i < payload->size; i++) {
+            uint8_t *tmp = payload->data;
+            dev->buf[index+i+1] = tmp[i];
+        }
+        /* count bytes */
+        index += payload->size;
+
+        /* next snip */
+        payload = payload->next;
+    }
+    dev->buf[0] = index+2; /* set packet size, reserve additional */
```

Comment:
again spaces - maybe you want to use [astyle](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions)